Modules/_io/fileio.c

```diff
@@ -178,6 +180,8 @@ _io_FileIO_close_impl(fileio *self, PyTypeObject *cls)
             PyErr_Clear();
         }
     }
+    PyMem_Free(self->stat_atopen);
+    self->stat_atopen = NULL;
```

Comment:
internal_close() already does the same, so it's redundant, no?

Reply:
Tradeoff for matching what `pyio` does (which does `os.close()` in these cases vs an `internal_close` that does the os close + other cleanup. Happy to drop though / definitely feels redundant to me in these two cases.